### ISSUE TITLE
Sounds Randomly Not Playing for Users on Headless Sessions

### ISSUE THREAD

**Issue by Enverex** [Neos-Metaverse/NeosPublic#2890]
### Describe the bug?

Since recently (cannot specify when exactly) users appear to be, at random, unable to hear sounds generated by OneShot audio clip players in headless sessions. This behaviour is very erratic though. This is what has been observed so far:

Even if a user (referred to in future as User A) is unable to hear sounds created from one specific OneShot generator, they'll still likely be able to hear sounds generated by other OneShot emitters. e.g. triggering the snoot boop on one player's avatar, vs another player's avatar. So the "problem" tends to be paired between a specific user and a specific OneShot generator, not all OneShots and not all players.

Other users still hear the sounds that user A generates by triggering something that fires OneShot audio, even if User A cannot hear them themselves.

Leaving and rejoining a session may fix the issue for that specific user. Though there's a chance that other OneShots may not work instead now.

### Relevant issues

None found.

### To Reproduce

Go to headless-hosted worlds and have a group of users all trigger OneShot sounds, see who does not hear them.

### Expected behavior

For everyone to hear triggered sounds.

### Log Files

I've not managed to experience this myself, I've just been around people that are now experiencing it in random worlds and also the Metamovie Project (e.g. the guns).

### How often does it happen?

Sometimes

### Does the bug persist after restarting Neos?

Yes

### Neos Version Number

2021.8.27.914

### What Platforms does this occur on?

Windows, Linux

### Link to Reproduction Item/World

Any headless session, doesn't seem to be specific to a sound or world.

### Did this work before?

Yes

### If it worked before, on which build?

Cannot determine

### Additional context

This is hard to narrow down as we can't seem to reproduce it reliably, people just randomly complain that these sounds cannot be heard which lead to the investigation.

### Reporters

Enverex#6528

**Comment by shadowpanther** [comment]
Okay, so what I was able to figure out (the symptoms). When a user loses their ability to hear oneshots:
- The affected user can hear oneshot if they trigger it (send the pulse)
- The affected user cannot hear oneshot if any other user trigger it, though they see the oneshot player appearing in the hierarchy
- Spawning a completely new Play One Shot node with a completely new Audio Clip Input and a completely new audio file plugged into it (done in iterations changing just one part each time) does not change anything (still hear their own triggered oneshots, still don't hear when triggered by anyone else)
- Another user spawning an audio file would produce an audio player orb that doesn't show/sound the playback for the affected user.

As this is currently only seen in headless sessions, one suspect change would be this:
```
- Headless now runs audio events at the actual rate they would fire at, rather than every frame
-- This can help reduce CPU usage in certain cases, particularly with higher tick rates
```
But I can't imagine how that would affect only one user at a time while everyone else are fine.

My current hypothesis on the combination of the symptoms would be - any audio player component created after the bug started and also created by the user other than the affected user would fail to play for the affected user. Need to test this further to confirm or debunk this.

**Comment by Frooxius** [comment]
This should be fixed by the master clock issue in 2021.8.29.1240. Please reopen if this still occurs!
